snapcraft/plugins/kernel.py

```diff
@@ -451,9 +459,15 @@ def _do_check_initrd(self, builtin, modules):
             logger.warn(warn)
 
     def pull(self):
-        super().pull()
-        snapcraft.download(
-            'core', 'stable', self.os_snap, self.project.deb_arch)
+        channel = self.options.kernel_channel
+        logger.info('Downloading core from the {!r} channel'.format(channel))
+        if channel in ['stable', 'beta', 'candidate', 'edge']:
```

Comment:
Note that there are hotfix branches as well, that can be created at any time.

Reply:
I can't tell if that's in favor of or against my suggestion. Is this list specific to the plugin or is any valid channel recognized by the store okay, and as such should be checked in the store API?

Reply:
Neither, it's in response to "these are all the channels, right?"

I sort of feel like we shouldn't be checking them here, but simply attempting to download the channel provided. That means this will work for hotfix branches as well, and also seems more pythonic because then we can simply catch any errors tossed by `snapcraft.download` instead of checking the channels first.

Reply:
Indeed, passing an invalid channel already raises an error:

...
Downloading core from the 'foobar' channel
Snap 'core' for 'amd64' cannot be found in the 'foobar' channel.

I guess, i can remove the channel checks from the kernel plugin and the unit tests too, since it's already handled by the download API.